src/libbson/CMakeLists.txt

```diff
@@ -248,10 +248,17 @@ set_target_properties (bson_shared PROPERTIES VERSION 0.0.0 SOVERSION 0)
 set_target_properties (bson_shared PROPERTIES OUTPUT_NAME "bson-${BSON_API_VERSION}" PREFIX "lib")
 
 if (ENABLE_APPLE_FRAMEWORK)
+
+   set (MONGOC_FRAMEWORK_BUNDLE_VERSION ${MONGOC_MAJOR_VERSION}.${MONGOC_MINOR_VERSION}.${MONGOC_MICRO_VERSION})
+   if ("${MONGOC_PRELEASE_VERSION}" STREQUAL "")
+      set (MONGOC_FRAMEWORK_BUNDLE_VERSION ${MONGOC_FRAMEWORK_BUNDLE_VERSION}${MONGOC_PRERELEASE_VERSION})
+   endif()
+
    set_target_properties(bson_shared PROPERTIES 
-      FRAMEWORK TRUE 
-      MACOSX_FRAMEWORK_BUNDLE_VERSION ${MONGOC_VERSION} 
-      MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${MONGOC_VERSION}
+      FRAMEWORK TRUE
+      FRAMEWORK_VERSION ${BSON_API_VERSION}
+      MACOSX_FRAMEWORK_BUNDLE_VERSION ${MONGOC_FRAMEWORK_BUNDLE_VERSION}
```

Comment:
If MACOSX_FRAMEWORK_BUNDLE_VERSION can include the prerelease version, then this could just be MONGOC_VERSION, right? Maybe I don't understand what you wrote in the PR comment, but this logic contradicts what you wrote there.

Reply:
The problem with just using MONGOC_VERSION for MACOSX_FRAMEWORK_BUNDLE_VERSION is that the `-` is disallowed. My work above basically tacks the prerelease version on sans `-` when it is set.